_source/_posts/2020-04-06-configure-okta-asa-for-gcp-jenkins.md

```diff
@@ -0,0 +1,856 @@
+
+{% img blog/asa-gcp-jenkins/gcp-menu.png alt:"GCP menu" width:"600" %}{: .center-image }
+
+- Open the left menu, and then go to **Compute Engine** > **VM instances**
+
+{% img blog/asa-gcp-jenkins/gcp-vm-instances-button.png alt:"GCP VM instances button" width:"600" %}{: .center-image }
+
+- You should land on the list of the VMs for that GCP project:
+
+{% img blog/asa-gcp-jenkins/gcp-vm-instances.png alt:"GCP VM instances" width:"600" %}{: .center-image }
+
+## Create Google Cloud VMs
+
+### gcp-ubuntu-bastion
+
+- Click on **Create Instance**
+
+{% img blog/asa-gcp-jenkins/create-instance.png alt:"create instance" width:"600" %}{: .center-image }
+
+- Enter a name for your instance: gcp-ubuntu-bastion
+
+- Select Machine type drop-down list: choose **micro**
+
+- Go to **Boot disk** > select **change** > select **Ubuntu 16.04 LTS**
+
+- Optionally, select Firewall http/https traffic (not required for bastion and target, but required for the Jenkins VM to view the web console)
+
+{% img blog/asa-gcp-jenkins/gcp-vm-config.png alt:"GCP vm config" width:"600" %}{: .center-image }
+
+- Click on **Create**
+
+### gcp-ubuntu-target
+
+Follow the same steps as above, and choose **gcp-ubuntu-target** as the VM name on Ubuntu 16.04 LTS.
+
+### gcp-jenkins
+
+Follow the same steps as above, but choose **gcp-jenkins** as the VM name on **Ubuntu 18.04 LTS** and machine type **g1-small** (1 vCPU, 1.7 GB memory).
+
+Then you will assign a static public IP for your VM, since you will want to load the Jenkins admin web panel, and bookmark that link in your web browser.
+
+- Reserve a static public IP and [assign it to your VM](https://cloud.google.com/compute/docs/ip-addresses/reserve-static-external-ip-address#reserve_new_static)
+
+{% img blog/asa-gcp-jenkins/gcp-attach.png alt:"GCP attach" width:"600" %}{: .center-image }
+
+- After 20 seconds, you should see your new static IP and verify that it's assigned to your VM:
+
+{% img blog/asa-gcp-jenkins/gcp-verify.png alt:"GCP verify" width:"600" %}{: .center-image }
+
+Note: if you already have a static IP to assign to your VM, read: [https://cloud.google.com/compute/docs/ip-addresses/reserve-static-external-ip-address#IP_assign](https://cloud.google.com/compute/docs/ip-addresses/reserve-static-external-ip-address#IP_assign)
+
+Then, you'll have to make that VM reachable from the internet over http 8080 to access the Jenkins web console:
+
+- Open your left side bar > **VPC network** > **Firewall rules**
+
+{% img blog/asa-gcp-jenkins/gcp-firewall-rules-button.png alt:"GCP firewall rules button" width:"600" %}{: .center-image }
+
+- Click on **Create Firewall Rule**
+
+{% img blog/asa-gcp-jenkins/gcp-create-firewall-rule-button.png alt:"GCP create firewall rule button" width:"600" %}{: .center-image }
+
+- Name the rule "inbound-http-8080" | Direction: Ingress | Incoming IP ranges: 0.0.0.0/0 | select tcp port 8080 | click on **Create**
```

Comment:
This doesn't look right.
<img width="898" alt="Screen Shot 2020-04-02 at 5 05 06 PM" src="https://user-images.githubusercontent.com/17892/78307840-1d4b1400-7504-11ea-9f68-ecf1f1a696b5.png">